// bsc-scan/dist/query.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var axios_1 = __importDefault(require("axios"));
var lodash_identity_1 = __importDefault(require("lodash.identity"));
var lodash_pickby_1 = __importDefault(require("lodash.pickby"));
var querystring_1 = __importDefault(require("querystring"));
var config_1 = __importDefault(require("./config"));
function query(queryOptions, requestConfig) {
    return __awaiter(this, void 0, void 0, function () {
        var address, contractAddress, txhash, module, action, _a, startBlock, _b, endBlock, _c, page, _d, offset, _e, sort, _f, apiKey, queryParams, _g, axiosConfig, rawAxiosResponse, query, response, _h, status, result;
        return __generator(this, function (_j) {
            switch (_j.label) {
                case 0:
                    address = queryOptions.address, contractAddress = queryOptions.contractAddress, txhash = queryOptions.txhash, module = queryOptions.module, action = queryOptions.action, _a = queryOptions.startBlock, startBlock = _a === void 0 ? 0 : _a, _b = queryOptions.endBlock, endBlock = _b === void 0 ? 99999999 : _b, _c = queryOptions.page, page = _c === void 0 ? undefined : _c, _d = queryOptions.offset, offset = _d === void 0 ? 10000 : _d, _e = queryOptions.sort, sort = _e === void 0 ? 'asc' : _e, _f = queryOptions.apiKey, apiKey = _f === void 0 ? config_1.default.apiKey : _f;
                    queryParams = lodash_pickby_1.default({
                        address: address,
                        contractAddress: contractAddress,
                        txhash: txhash,
                        module: module,
                        action: action,
                        startBlock: startBlock,
                        endBlock: endBlock,
                        page: page,
                        offset: offset,
                        sort: sort,
                        apiKey: apiKey,
                    }, lodash_identity_1.default);
                    _g = requestConfig || {}, axiosConfig = _g.axiosConfig, rawAxiosResponse = _g.rawAxiosResponse;
                    query = querystring_1.default.stringify(queryParams);
                    return [4 /*yield*/, axios_1.default.get(config_1.default.url + "/api?" + query, axiosConfig)];
                case 1:
                    response = _j.sent();
                    if (rawAxiosResponse) {
                        return [2 /*return*/, response];
                    }
                    _h = response.data, status = _h.status, result = _h.result;
                    if (status === '0') {
                        throw new Error(String(result));
                    }
                    return [2 /*return*/, result];
            }
        });
    });
}
exports.default = query;
