// bsc-scan/dist/token.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var query_1 = __importDefault(require("./query"));
function getTokenSupplyByContractAddress(contractAddress, requestConfig) {
    return query_1.default({
        contractAddress: contractAddress,
        module: 'stats',
        action: 'tokensupply',
    }, requestConfig);
}
function getCirculatingSupplyByContractAddress(contractAddress, requestConfig) {
    return query_1.default({
        contractAddress: contractAddress,
        module: 'stats',
        action: 'tokenCsupply',
    }, requestConfig);
}
function getAccountBalanceForTokenContractAddress(address, contractAddress, requestConfig) {
    return query_1.default({
        address: address,
        contractAddress: contractAddress,
        module: 'account',
        action: 'tokenbalance',
        tag: 'latest',
    }, requestConfig);
}
function getHistoricalAccountBalanceForTokenContractAddress(address, contractAddress, blockno, requestConfig) {
    return query_1.default({
        address: address,
        contractAddress: contractAddress,
        module: 'account',
        action: 'tokenbalancehistory',
        blockno: blockno,
        tag: 'latest',
    }, requestConfig);
}
exports.default = {
    getTokenSupplyByContractAddress: getTokenSupplyByContractAddress,
    getCirculatingSupplyByContractAddress: getCirculatingSupplyByContractAddress,
    getAccountBalanceForTokenContractAddress: getAccountBalanceForTokenContractAddress,
    getHistoricalAccountBalanceForTokenContractAddress: getHistoricalAccountBalanceForTokenContractAddress,
};
